playback/src/audio_backend/pulseaudio.rs

```diff
@@ -3,82 +3,126 @@ use crate::config::AudioFormat;
 use crate::convert::Converter;
 use crate::decoder::AudioPacket;
 use crate::{NUM_CHANNELS, SAMPLE_RATE};
-use libpulse_binding::{self as pulse, stream::Direction};
+use libpulse_binding::{self as pulse, error::PAErr, stream::Direction};
 use libpulse_simple_binding::Simple;
 use std::io;
+use thiserror::Error;
 
 const APP_NAME: &str = "librespot";
 const STREAM_NAME: &str = "Spotify endpoint";
 
+#[derive(Debug, Error)]
+enum PulseError {
+    #[error("Error starting PulseAudioSink, invalid PulseAudio sample spec")]
+    InvalidSampleSpec,
+    #[error("Error starting PulseAudioSink, could not connect to PulseAudio server, {0}")]
+    ConnectionRefused(PAErr),
+    #[error("Error stopping PulseAudioSink, failed to drain PulseAudio server buffer, {0}")]
+    DrainFailure(PAErr),
+    #[error("Error in PulseAudioSink, Not connected to PulseAudio server")]
+    ServerNone,
```

Comment:
Why not plain & easy `NotConnected`?

Reply:
So `ServerNone` > `NotConnected` Ok. I was just going for the whole parity of errors you were after. `NotConnected` makes no sense for `alsa`.

Reply:
I don't understand what you mean by parity in this case, because I don't think that there a `ServerNone` for Alsa?

Reply:
I was going to do `PcmNone`.

Reply:
Done the way you wanted.